src/test/java/bio/terra/service/snapshot/SnapshotServiceTest.java

```diff
@@ -1203,6 +1207,51 @@ void makeSnapshotFromSnapshotRequestByRequestId() {
     assertThat(actual.getCreationInformation(), is(contentsModel));
   }
 
+  @Test
+  void getSourceDatasetsFromSnapshotRequestHandlesByRequestId() {
+    UUID snapshotAccessRequestId = UUID.randomUUID();
+    SnapshotRequestIdModel requestIdModel =
+        new SnapshotRequestIdModel().snapshotRequestId(snapshotAccessRequestId);
+    SnapshotRequestContentsModel contentsModel =
+        new SnapshotRequestContentsModel()
+            .datasetName("datasetName")
+            .mode(SnapshotRequestContentsModel.ModeEnum.BYREQUESTID)
+            .requestIdSpec(requestIdModel);
+    SnapshotRequestModel snapshotRequestModel =
+        new SnapshotRequestModel().contents(List.of(contentsModel));
+
+    SnapshotAccessRequestResponse snapshotAccessRequest =
+        new SnapshotAccessRequestResponse().sourceSnapshotId(snapshotId);
+    Dataset dataset = new Dataset().id(datasetId);
+    Snapshot snapshot =
+        new Snapshot().snapshotSources(List.of(new SnapshotSource().dataset(dataset)));
+
+    when(snapshotRequestDao.getById(snapshotAccessRequestId)).thenReturn(snapshotAccessRequest);
+    when(snapshotDao.retrieveSnapshot(snapshotId)).thenReturn(snapshot);
+
+    List<Dataset> datasets = service.getSourceDatasetsFromSnapshotRequest(snapshotRequestModel);
+
+    assertThat(datasets.get(0), is(dataset));
+  }
+
+  @Test
+  void getSourceDatasetsFromSnapshotRequestHandlesNonByRequestId() {
+    String datasetName = "datasetName";
```

Comment:
since this is also used in the above test, maybe you could make it a field variable